packages/kit/src/exports/node/index.js

```diff
@@ -146,8 +146,7 @@ export async function setResponse(res, response) {
 				key,
 				key === 'set-cookie'
 					? set_cookie_parser.splitCookiesString(
-							// This is absurd but necessary, TODO: investigate why
-							/** @type {string}*/ (response.headers.get(key))
+							value
```

Comment:
Mhhhm @gtm-nayan do you remember why exactly this was necessary? You added it in #9638 - want to make sure that not just using the value really was on purpose

Reply:
Will have to test again to be exact but IIRC, there can be multiple set-cookie headers and how the value is returned with multiple of them, didn't work with how we use it here.